packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1409,3 +1409,77 @@ class SliverConstrainedCrossAxis extends SingleChildRenderObjectWidget {
     renderObject.maxExtent = maxExtent;
   }
 }
+
+/// Set a flex factor for allocating space in the cross axis direction.
+///
+/// This is a [ParentDataWidget] to be used in [SliverCrossAxisGroup].
+/// The remaining space (cross axis extent of the parent minus the extent of
+/// children with constrained cross axis) will be divided according to the flex values.
+class SliverExpanded extends ParentDataWidget<SliverPhysicalContainerParentData> {
+  /// Creates an object that assigns a [flex] value to the child sliver.
+  ///
+  /// The provided [flex] value must be at least 0.
+  const SliverExpanded({
+    super.key,
+    required this.flex,
+    required Widget sliver,
+  }): assert(flex > 0 && flex < double.infinity),
+      super(child: sliver);
+
+  /// Flex value for allocating cross axis extent left after laying out the children with
+  /// constrained cross axis. The children with flex values will have the remaining extent
+  /// allocated proportionally to their flex value. This must an integer between
+  /// 0 and infinity, exclusive.
```

Comment:
Well, I did add an assertion to make sure that it's less than `double.infinity` (which I assume means finite), but I think the behavior will just become weird if we allow this (since the infinite flex sliver will take up all remaining space and the rest is allotted 0).